spec/bolt_server/transport_app_spec.rb

```diff
@@ -38,288 +38,255 @@ def file_data(file)
-        ex = RuntimeError.new("oops")
-        ex.set_backtrace('/path/to/bolt/code.rb:42')
-        Bolt::Result.from_exception(conn_target('winrm'), ex)
+        result = last_response.body
+        expect(result).to match(%r{The property '#/target' did not contain a required property of 'password'})
       end
 
-      it 'scrubs stack trace from result' do
-        allow_any_instance_of(Bolt::Executor).to receive(:batch_execute).and_return(Bolt::ResultSet.new([error_result]))
-        body = {
-          'task': echo_task,
-          'target': {
-            'hostname': target[:host],
-            'user': target[:user],
-            'password': target[:password],
-            'port': target[:port]
-          },
-          'parameters': { "input": "Hello!" }
-        }
+      it 'fails if either port or connect-timeout is a string' do
+        body = { target: {
+          hostname: target[:host],
+          uaser: target[:user],
+          password: target[:password],
+          port: 'port',
+          'connect-timeout': 'timeout'
+        } }
+
+        post(path, JSON.generate(body), 'CONTENT_TYPE' => 'text/json')
+
+        expect(last_response).not_to be_ok
+        expect(last_response.status).to eq(400)
+
+        result = last_response.body
+        expect(result).to match(%r{The property '#/target/port' of type string did not match the following type: integer})
+        expect(result).to match(%r{The property '#/target/connect-timeout' of type string did not match the following type: integer})
+      end
+
+      it 'performs the action and scrubs any stack traces from the result' do
+        body = { target: {
+          hostname: target[:host],
+          user: target[:user],
+          password: target[:password],
+          port: target[:port]
+        } }
+
+        expect_any_instance_of(BoltServer::TransportApp)
+          .to receive(:scrub_stack_trace).with(result.status_hash).and_return({})
+
+        post(path, JSON.generate(body), 'CONTENT_TYPE' => 'text/json')
 
-        post path, JSON.generate(body), 'CONTENT_TYPE' => 'text/json'
         expect(last_response).to be_ok
         expect(last_response.status).to eq(200)
-        result = JSON.parse(last_response.body)
-        expect(result).to include('status' => 'failure')
-        expect(result['result']['_error']).to include('msg' => 'oops', 'details' => { 'class' => 'RuntimeError' })
-        expect(result['result']['_error']).not_to include('stack_trace')
       end
     end
   end
 
-  context 'when raising errors' do
-    let(:target) { conn_info('ssh') }
-    let(:echo_task) {
-      {
-        'name': 'echo',
-        'metadata': {
-          'description': 'Echo a message',
-          'parameters': { 'message': 'Default message' }
-        },
-        'file': {
-          'file_content': Base64.encode64("#!/usr/bin/env bash\necho $PT_message"),
-          'filename': "echo.sh"
-        }
-      }
-    }
+  describe 'action endpoint' do
```

Comment:
My intent is to use this section for testing of other endpoints like run_command, run_script, or upload_file.